app/main/views/direct_award.py

```diff
@@ -193,47 +207,50 @@ def create_project_search(project_id):
     return jsonify(search=search.serialize()), 201
 
 
-@main.route('/direct-award/projects/<int:project_id>/searches/<int:search_id>', methods=['GET'])
-def get_project_search(project_id, search_id):
+@main.route('/direct-award/projects/<int:project_external_id>/searches/<int:search_id>', methods=['GET'])
+def get_project_search(project_external_id, search_id):
     # If the project doesn't exist, we should 404 early.
-    DirectAwardProject.query.filter(DirectAwardProject.id == project_id).first_or_404()
+    project = get_project_by_id(project_external_id)
 
     search = DirectAwardSearch.query.filter(
         DirectAwardSearch.id == search_id,
-        DirectAwardSearch.project_id == project_id
+        DirectAwardSearch.project_id == project.id
     ).first_or_404()
 
     return jsonify(search=search.serialize())
 
 
-@main.route('/direct-award/projects/<int:project_id>/services', methods=['GET'])
-def list_project_services(project_id):
+@main.route('/direct-award/projects/<int:project_external_id>/services', methods=['GET'])
+def list_project_services(project_external_id):
     """This endpoint returns all the services associated with a particular (locked) Direct Award Project. It returns
     some fairly arbitrary data which is obviously not ideal, but as we don't have a task runner that we can utilise to
     manage jobs like this and we need it to be responsive on-click, we're having to jerryrig in some data extraction and
     transformation in here. Specifically we are returning supplier name and contact information, and returning from
     the service data JSON keys as specified in the request (which will be loaded via a framework-specific manifest)."""
     page = get_valid_page_or_1()
     requested_fields = request.args.get('fields', '').split(',')
-    project = DirectAwardProject.query.filter(DirectAwardProject.id == project_id).first_or_404()
+    project = get_project_by_id(project_external_id)
 
     if not project.locked_at:
-        abort(400, 'Project has not been locked: {}'.format(project_id))
+        abort(400, 'Project has not been locked: {}'.format(project.id))
 
     # TODO: This should work for _all_ active searches, not just the first (although we currently enforce only one
     # TODO: active search) - SW 21/09/2017
     search = DirectAwardSearch.query.filter(
-        DirectAwardSearch.project_id == project_id,
+        DirectAwardSearch.project_id == project.id,
         DirectAwardSearch.active == True  # noqa
     ).first()
     if not search:
-        abort(400, 'Project does not have a saved search: {}'.format(project_id))
+        abort(400, 'Project does not have a saved search: {}'.format(project.id))
 
     paginated_archived_services = search.archived_services.paginate(
         page=page,
         per_page=current_app.config['DM_API_PROJECTS_PAGE_SIZE'],
     )
 
+    # Serializing gives us the ID that is intended to leave the API.
+    project_id = project.serialize()['id']
```

Comment:
Do we need to do this? This is the external ID right? Can't we just reference the external ID in the archived service instead?

Reply:
This will return the external ID, yes. It's not strictly necessary, and I'm open to being convinced otherwise. My justification was simply that serialize() then becomes the only place where you have a direct reference to 'external_id', but maybe hiding this fact just makes it more confusing.

I think it probably does in reality. I'll change it.